pandas/core/frame.py

```diff
@@ -2604,6 +2604,177 @@ def to_html(
             render_links=render_links,
         )
 
+    def to_xml(
+        self,
+        path_or_buffer: Optional[FilePathOrBuffer[str]] = None,
+        index: bool = True,
+        root_name: Optional[str] = "data",
+        row_name: Optional[str] = "row",
+        na_rep: Optional[str] = None,
+        attr_cols: Optional[Union[str, List[str]]] = None,
+        elem_cols: Optional[Union[str, List[str]]] = None,
+        namespaces: Optional[Dict[Optional[str], str]] = None,
+        prefix: Optional[str] = None,
+        encoding: str = "utf-8",
+        xml_declaration: Optional[bool] = True,
+        pretty_print: Optional[bool] = True,
+        parser: Optional[str] = "lxml",
+        stylesheet: Optional[FilePathOrBuffer[str]] = None,
+    ) -> Optional[str]:
+        """
+        Render a DataFrame to an XML document.
+
+        .. versionadded:: 1.3.0
+
+        Parameters
+        ----------
+        path_or_buffer : str, path object or file-like object, optional
+            File to write output to. If None, the output is returned as a
+            string.
+        index : bool, default True
+            Whether to include index in XML document.
+        root_name : str, default 'data'
+            The name of root element in XML document.
+        row_name : str, default 'row'
+            The name of row element in XML document.
+        na_rep : str, optional
+            Missing data representation.
+        attr_cols : list-like, optional
+            List of columns to write as attributes in row element.
+            Hierarchical columns will be flattened with underscore
+            delimiting the different levels.
+        elem_cols : list-like, optional
```

Comment:
are *all* of these options really useful / necessary? yes you are defaulting lots of things, but this seems pretty complicated. on what to use when.

Reply:
I mirrored `to_xml` with `read_xml` so include element/attribute handling and namespaces where `to_xml` outputs can be read back in with `read_xml`. As noted in io.rst, there is no standard XML. Some users may need the default element-centric output and others attribute-centric and still others a mix of both depending on needs. Some schemas like KML require a default namespace. Root and row names, declaration and pretty print are helpers for quick redesign similar to the formatting params in `to_html`. While these can be trivially handled in `stylesheet`, many do not know XSLT used for more extensive transformations.